test/indexes.spec.js

```diff
@@ -356,6 +355,166 @@ describe('Model index tests', function () {
       expect(actualRefs).to.deep.equal(expected);
     });
 
+    it('should create index values for nested values', function () {
+      var userSchema = new lounge.Schema({
+        firstName: String,
+        lastName: String,
+        email: String,
+        company: {
+          name: {
+            type: String,
+            index: true,
+            indexName: 'company',
+            indexType: 'array'
+          }
+        }
+      });
+
+      var User = lounge.model('User', userSchema);
+
+      expect(User.findByCompany).to.be.ok;
+      expect(User.findByCompany).to.be.an.instanceof(Function);
+
+      var data = {
+        firstName: 'Joe',
+        lastName: 'Smith',
+        email: 'jsmith@acme.inc',
+        company: { name: 'Acme Inc' }
+      };
+
+      var expected = {
+        'company': {
+          path: 'company.name',
+          value: 'Acme Inc',
+          name: 'company',
+          indexType: 'array'
+        }
+      };
+
+      var actualRefs = couchUtil.buildRefValues(userSchema.indexes, data);
+      expect(actualRefs).to.deep.equal(expected);
+    });
+
+    it('should create index values for nested values through index function', function () {
+      var userSchema = new lounge.Schema({
+        firstName: String,
+        lastName: String,
+        email: String,
+        company: {
+          name: {
+            type: String
+          }
+        }
+      });
+
+      userSchema.index('company.name', {indexName: 'company', indexType: 'array'});
+
+      var User = lounge.model('User', userSchema);
+
+      expect(User.findByCompany).to.be.ok;
+      expect(User.findByCompany).to.be.an.instanceof(Function);
+
+      var data = {
+        firstName: 'Joe',
+        lastName: 'Smith',
+        email: 'jsmith@acme.inc',
+        company: { name: 'Acme Inc' }
+
+      };
+
+      var expected = {
+        'company': {
+          path: 'company.name',
+          value: 'Acme Inc',
+          name: 'company',
+          indexType: 'array'
+        }
+      };
+
+      var actualRefs = couchUtil.buildRefValues(userSchema.indexes, data);
+      expect(actualRefs).to.deep.equal(expected);
+    });
+
+    it('should create compound index values for nested values', function () {
+      var userSchema = new lounge.Schema({
+        firstName: String,
+        lastName: String,
+        email: String,
+        company: {
+          name: String
+        }
+      });
+
+      userSchema.index(['email', 'company.name'], { indexName: 'company', indexType: 'array' });
```

Comment:
do we have a test for similar scenario, but not setting indexName? so that it would result in `User.findByEmailAndCompany` or something?